### DISCUSSION TITLE
[Feature]: 前端修改配置

### DISCUSSION THREAD

**Discussion by sysfox** [walinejs/waline#1531] [Ideas]
### 功能概述 | Describe the feature

能否像twikoo那样，将各项配置(即各个环境变量)存储在数据库中，想要修改配置就在前端面板中进行修改(专门开个登陆页只能修改个人信息和管理评论也太大材小用了吧)，不然每次修改都要重新部署

**Comment by lizheming** [comment]
Waline 是前后端分离的项目，用户是可以自定义前端组件的，官方只是提供了一个默认的客户端。不同的客户端的配置可能不一样，如果写在后台的话，有种和官方默认客户端绑定的感觉。

**Comment by Mister-Hope** [comment]
就我们希望看到的是用户可以选择多种客户端，或者为自己的客户端使用多种风格。

从目前的角度看，包括预览、图片、Gif搜索、表情等都是纯客户端的功能，前端负责提供这些功能并生成对应的评论数据直接上报给后端，我们也希望看到有第三方客户端实现更多精彩功能或一些定制化的特殊能力，以及一些为官方客户端添加新功能的 PR。

所以从技术的角度来讲，客户端配置也仅仅是客户端配置，与后台有关的只有 serverURL 地址，以及通过标准 API 实行通信与操作而已。所以我个人也同样不支持在服务器配置客户端的 idea。

**Reply by sysfox** [reply]
我指的是能否将服务端的一些环境变量（如安全域名，表情列表，邮件通知等设置）转移到
serverURL/ui 里进行管理，这样就不用改一次变量就重新部署一次了

**Reply by lizheming** [reply]
暂时没有这个计划，这种一次性配置建议还是在环境变量里配比较好。一是有些配置比如数据库的配置是一定要在环境变量中配的，如果修改后会有配置割裂，二是如果安全域名这种配置放到数据库之后，所有的请求都会增加一次请求，对接口性能以及一些数据库的免费额度也会有一定影响。

**Reply by sysfox** [reply]
那就把一些没有冲突的配置加入，毕竟像数据库链接、安全域名这些配置管理员都是很少更改的

**Comment by sysfox** [comment]
@lizheming  @Mister-Hope
